xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -739,8 +739,17 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterStringListProperty)
-  #set ($whereSql = "${whereSql} and concat('|', concat(upper(${safe_tableAlias}.textValue), '|')) like upper(?)")
-  #set ($discard = $whereParams.add("%|$filterValue|%"))
+  #set ($filterValues = $request.getParameterValues($colname))
+  #set ($joinOperator = "$!{request.get(""${colname}${joinModeMarker}"").toUpperCase()}")
+  #if ($joinOperator != 'AND' &amp;&amp; $joinOperator != 'OR')
+    #set ($joinOperator = 'AND')
+  #end
+  #set ($filterQuery = "concat('|', concat(upper(${safe_tableAlias}.textValue), '|')) LIKE upper(?)")
+  #set ($constraints = $stringtool.repeat($filterQuery, $joinOperator, $filterValues.size()))
```

Comment:
There's no white space around the join operator. Isn't this a problem? Don't you get "ANDconcat(..."?